### PR TITLE
Event occurrence filters

### PR DESCRIPTION
**Pull request by luisa-li**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/79)
Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

# How Has This Been Tested?
Please describe the tests that you manually ran to verify your changes (beyond any unit/integration tests written and ran).

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)
Screenshot 2026-02-17 at 11 52 47 AM
Screenshot 2026-02-17 at 11 52 28 AM
Screenshot 2026-02-17 at 11 52 08 AM
Screenshot 2026-02-17 at 11 50 52 AM
Screenshot 2026-02-17 at 11 48 49 AM

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: backend/internal/storage/postgres/schema/event-occurrence/sql/get_all.sql [hunk 1]
```diff
@@ -36,5 +36,19 @@ SELECT
...
+
+WHERE 1=1
+AND ($3::text IS NULL OR e.title ILIKE '%' || $3 || '%' OR e.description ILIKE '%' || $3 || '%')
+AND ($4::int IS NULL OR EXTRACT(EPOCH FROM (eo.end_time - eo.start_time))/60 >= $4)
+AND ($5::int IS NULL OR EXTRACT(EPOCH FROM (eo.end_time - eo.start_time))/60 <= $5)
+AND (
+    $6::float IS NULL 
+    OR $7::float IS NULL
+    OR $8::float IS NULL
+    OR earth_distance(
```
Comment (adescoteaux1) [inline backend/internal/storage/postgres/schema/event-occurrence/sql/get_all.sql]: ooooo earth_distance, new func unlocked

File: backend/internal/service/routes/event_occurrences_routes.go [hunk 2]
```diff
@@ -34,7 +36,46 @@ func SetupEventOccurrencesRoutes(api huma.API, repo *storage.Repository) {
...
+		if input.PriceTier != "" && !validTiers[input.PriceTier] {
+			return nil, huma.Error400BadRequest("price must be one of $, $$, $$$")
+		}
+
+		// map to DB-level filter object, using pointers internally
+		var filters models.GetAllEventOccurrencesFilter
+		if input.Search != "" {
+			filters.Search = &input.Search
+		}
+		if input.Latitude != 0 && input.Longitude != 0 && input.RadiusKm != 0 {
```
Comment (adescoteaux1) [inline backend/internal/service/routes/event_occurrences_routes.go]: i think 0 is a valid lat/long value, so this might not be a good check

Comment (luisa-li) [reply backend/internal/service/routes/event_occurrences_routes.go]: fixed this so a 0 as a lat long is taken as a valid filter, but not giving it means unrestricted.
